bindings/rust/src/bindings/mod.rs

```diff
@@ -102,12 +100,38 @@ impl KZGSettings {
     /// FIELD_ELEMENT_PER_BLOB g1 byte values
     /// 65 g2 byte values
     pub fn load_trusted_setup_file(file_path: PathBuf) -> Result<Self, Error> {
-        let file_path = CString::new(file_path.as_os_str().as_bytes()).map_err(|e| {
-            Error::InvalidTrustedSetup(format!("Invalid trusted setup file: {:?}", e))
-        })?;
+        let mode = unsafe { CString::from_vec_unchecked(vec!['r' as u8]) };
+        #[cfg(unix)]
+        let file_ptr = {
+            use std::os::unix::prelude::OsStrExt;
+
+            let file_path = CString::new(file_path.as_os_str().as_bytes()).map_err(|e| {
+                Error::InvalidTrustedSetup(format!("Invalid trusted setup file: {:?}", e))
+            })?;
+
+            let fp = unsafe { libc::fopen(file_path.as_ptr(), mode.as_ptr()) };
+            if fp.is_null() {
+                let e = std::io::Error::last_os_error();
+                println!("Failed to get file_descriptor {e}");
+                return Err(Error::CError(C_KZG_RET::C_KZG_BADARGS));
+            }
+            fp
+        };
+        #[cfg(windows)]
+        let file_ptr = {
+            let file_path =
+                CString::new(file_path.as_os_str().to_str().unwrap().as_bytes()).unwrap();
+
+            let fp = unsafe { libc::fopen(file_path.as_ptr(), mode.as_ptr()) };
+            if fp.is_null() {
+                let e = std::io::Error::last_os_error();
+                println!("Failed to get file_descriptor {e}");
+                return Err(Error::CError(C_KZG_RET::C_KZG_BADARGS));
+            }
+            fp
```

Comment:
I think this chunk is the exact same as unix version. Is it possible to de-dup a bit here?